Fix the sunpy test suite to not error if packages are missing

<!-- This comments are hidden when you submit the issue so you do not need to remove them!
Please be sure to check out our contributing guidelines: https://github.com/sunpy/sunpy/blob/master/CONTRIBUTING.rst
Please be sure to check out our code of conduct:
https://github.com/sunpy/sunpy/blob/master/CODE_OF_CONDUCT.rst -->

<!-- Please have a search on our GitHub repository to see if a similar issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied by the resolution.
If not please go ahead and open an issue! -->

### Description
<!-- Provide a general description of the bug. -->
I just had to reinstall my conda and python installations (anaconda3, python3.7), along with a lot of packages I need, including sunpy. I just installed it and ran sunpy.self_test(), and got several errors.

### Expected behavior
<!-- What did you expect to happen. -->
sunpy.self_test() should run with no errors.

### Actual behavior
<!-- What actually happened. -->
<!-- Was the output confusing or poorly described? -->
The install was missing a couple dependencies (hypothesis and asdf). After installing the missing modules, I am getting more errors regarding the `markers` configuration options.

(conda install flaky gets rid of a few more errors, but several still occur.)

Here's the initial self_test() output I got:

```
========================================================================================= test session starts ==========================================================================================
platform linux -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
rootdir: /my/dir/anaconda3, inifile: setup.cfg
plugins: openfiles-0.3.2, remotedata-0.3.1, arraydiff-0.3, doctestplus-0.3.0
collected 1304 items / 15 errors / 1 skipped / 1288 selected                                                                                                                                           

================================================================================================ ERRORS ================================================================================================
______________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/coordinates/tests/test_offset_frame.py _______________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/coordinates/tests/test_offset_frame.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/coordinates/tests/test_offset_frame.py:2: in <module>
    import hypothesis.strategies as st
E   ModuleNotFoundError: No module named 'hypothesis'
___________________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/database/tests/test_tables.py ___________________________________________________________
'flaky' not found in `markers` configuration option
_____________________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/instr/tests/test_fermi.py _____________________________________________________________
'flaky' not found in `markers` configuration option
_____________________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/instr/tests/test_goes.py ______________________________________________________________
'array_compare' not found in `markers` configuration option
_______________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/io/special/asdf/tags/tests/test_coordinate_frames.py ________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/io/special/asdf/tags/tests/test_coordinate_frames.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/io/special/asdf/tags/tests/test_coordinate_frames.py:8: in <module>
    from asdf.tests.helpers import assert_roundtrip_tree
E   ModuleNotFoundError: No module named 'asdf'
____________________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/map/tests/test_plotting.py _____________________________________________________________
'figure' not found in `markers` configuration option
_____________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_fermi_gbm.py _____________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_fermi_gbm.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_fermi_gbm.py:10: in <module>
    from sunpy.net.tests.strategies import time_attr
../anaconda3/lib/python3.7/site-packages/sunpy/net/tests/strategies.py:7: in <module>
    import hypothesis.strategies as st
E   ModuleNotFoundError: No module named 'hypothesis'
______________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_goes_ud.py ______________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_goes_ud.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_goes_ud.py:2: in <module>
    from hypothesis import given, example, settings
E   ModuleNotFoundError: No module named 'hypothesis'
______________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_lyra_ud.py ______________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_lyra_ud.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_lyra_ud.py:12: in <module>
    from hypothesis import given, settings
E   ModuleNotFoundError: No module named 'hypothesis'
_______________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_norh.py ________________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_norh.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/net/dataretriever/tests/test_norh.py:14: in <module>
    from hypothesis import given, settings
E   ModuleNotFoundError: No module named 'hypothesis'
______________________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/net/tests/test_fido.py _______________________________________________________________
ImportError while importing test module '/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/net/tests/test_fido.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../anaconda3/lib/python3.7/site-packages/sunpy/net/tests/test_fido.py:6: in <module>
    import hypothesis.strategies as st
E   ModuleNotFoundError: No module named 'hypothesis'
______________________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/net/tests/test_helio.py ______________________________________________________________
'flaky' not found in `markers` configuration option
____________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/physics/tests/test_differential_rotation.py ____________________________________________________
'array_compare' not found in `markers` configuration option
______________________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/timeseries/tests/test_timeseriesbase.py ______________________________________________________
'figure' not found in `markers` configuration option
_______________________________________________ ERROR collecting lib/python3.7/site-packages/sunpy/visualization/animator/tests/test_basefuncanimator.py _______________________________________________
'figure' not found in `markers` configuration option
=========================================================================================== warnings summary ===========================================================================================
/my/dir/anaconda3/lib/python3.7/site-packages/pywt/_utils.py:6
  /my/dir/anaconda3/lib/python3.7/site-packages/pywt/_utils.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Iterable

/my/dir/anaconda3/lib/python3.7/site-packages/zeep/loader.py:3
  /my/dir/anaconda3/lib/python3.7/site-packages/zeep/loader.py:3: DeprecationWarning: defusedxml.lxml is no longer supported and will be removed in a future release.
    from defusedxml.lxml import fromstring

/my/dir/anaconda3/lib/python3.7/site-packages/html5lib/_trie/_base.py:3
  /my/dir/anaconda3/lib/python3.7/site-packages/html5lib/_trie/_base.py:3: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/map/mapbase.py:176
  /my/dir/anaconda3/lib/python3.7/site-packages/sunpy/map/mapbase.py:176: SunpyUserWarning: This file contains more than 2 dimensions. Data will be truncated to the first two dimensions.
    "Data will be truncated to the first two dimensions.", SunpyUserWarning)

/my/dir/anaconda3/lib/python3.7/site-packages/sunpy/map/map_factory.py:292
  /my/dir/anaconda3/lib/python3.7/site-packages/sunpy/map/map_factory.py:292: UserWarning: One of the data, header pairs failed to validate with: Image coordinate units for axis 1 not present in metadata.
  Image coordinate units for axis 2 not present in metadata.
  See https://docs.sunpy.org/en/stable/code_ref/map.html#fixing-map-metadata` for instructions on how to add missing metadata.
    warnings.warn(f"One of the data, header pairs failed to validate with: {e}")

/my/dir/anaconda3/lib/python3.7/site-packages/astropy/io/fits/file.py:379
  /my/dir/anaconda3/lib/python3.7/site-packages/astropy/io/fits/file.py:379: AstropyUserWarning: File may have been truncated: actual file length (538688) is smaller than the expected size (541440)
    .format(self.size, pos), AstropyUserWarning)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 15 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=========================================================================== 1 skipped, 6 warnings, 15 error in 15.52 seconds ===========================================================================
<ExitCode.INTERRUPTED: 2>

```

### Steps to Reproduce
<!-- Ideally a code example could be provided so we can run it ourselves. -->
<!-- If you are pasting code, use tripe backticks (```) around your code snippet. -->

1. reinstall conda/python...
2. ```conda config --add channels conda-forge```
3. ```conda install sunpy```
4. ```python```
5. ```import sunpy```
6. ```sunpy.self_test()```

Again, ```conda install hypothesis asdf flaky``` gets rid of several errors, but not all of them.

### System Details
<!-- We at least need to know the SunPy version you are using. -->
<!-- We provide a short function in SunPy that will provide some of the below information. -->
<!-- It is sunpy.util.system_info(), this is optional but strongly recommended. -->

  - SunPy Version: 1.0.2
  - Astropy Version: 3.2.1
  - Python Version: 3.7.3
  - OS information: CentOS Linux 7.3.1611 Core

--- a/docs/guide/installation.rst
+++ b/docs/guide/installation.rst
@@ -5,7 +5,7 @@ Installation
 Requirements
 ============
 
-sunpy enforces a minimum Python version that you will have to use and currently that is Python 3.7 or higher.
+sunpy requires Python 3.7 or higher.
 
 Installing Scientific Python and sunpy
 ======================================
@@ -47,43 +47,11 @@ You can update to the latest version by running::
 
     conda update sunpy
 
-Testing sunpy
-=============
-
-sunpy provides a method to run the basic test suite that will check that the install has worked correctly.
-
-To do so, use the :func:`sunpy.self_test`::
-
-    import sunpy
-    sunpy.self_test()
-
-You will see something like the following in your terminal::
-
-    Starting sunpy self test...
-    Checking for packages needed to run sunpy:
-    All required and optional sunpy dependencies are installed.
-    Starting the sunpy test suite:
-    ...
-
-The tests will run and will report any fails which you can report at the `sunpy issue tracker <https://github.com/sunpy/sunpy/issues>`__ and we will strive to help.
-
-It is possible to run this command and in a situation where not all packages are installed in order to run the test suite, you will see::
-
-    Starting sunpy self test...
-    Checking for packages needed to run sunpy:
-    The following packages are not installed for the sunpy[database] requirement:
-    * sqlalchemy
-    ...
-    You do not have all the required dependencies installed to run the sunpy test suite.
-    If you want to run the sunpy tests install the 'tests' extra with `pip install sunpy[all,tests]`
-
-This should not appear under a conda install of sunpy but can under a pip install of sunpy.
-This does not mean sunpy is broken, you will need to install the extra packages to ensure a "complete" installation of sunpy.
-
 Installing sunpy on top of an existing scientific Python environment
 --------------------------------------------------------------------
 
-This section assumes you already have everything setup whether that be Conda or a Python virtual environment and that the commands are exceucted within these.
+This section assumes you already have everything setup, whether that be conda or a Python virtual environment.
+These commands are to be executed within these environments.
 
 Conda
 ^^^^^
@@ -128,3 +96,36 @@ If you want to develop sunpy we would strongly recommend reading the `Newcomers'
     This error implies you have an incorrectly configured virtual environment or it is not activated.
 
     If you really do not want to use any virtual environment, you can always do ``pip install --user sunpy``.
+
+Testing sunpy
+=============
+
+sunpy provides a method to run the basic test suite that will check that the install has worked correctly.
+
+To run the basic test suite and ensure that your sunpy install is working correctly, use the :func:`sunpy.self_test`::
+
+    import sunpy
+    sunpy.self_test()
+
+You will see something like the following in your terminal::
+
+    Starting sunpy self test...
+    Checking for packages needed to run sunpy:
+    All required and optional sunpy dependencies are installed.
+    Starting the sunpy test suite:
+    ...
+
+    The tests will run and will report any fails.  You can report these through the `sunpy issue tracker <https://github.com/sunpy/sunpy/issues>`__ and we will strive to help.
+
+It is possible to run this command in a situation where not all packages are installed. If this is the case, you will see the following when you run the test suite::
+
+    Starting sunpy self test...
+    Checking for packages needed to run sunpy:
+    The following packages are not installed for the sunpy[database] requirement:
+    * sqlalchemy
+    ...
+    You do not have all the required dependencies installed to run the sunpy test suite.
+    If you want to run the sunpy tests install the 'tests' extra with `pip install sunpy[all,tests]`
+
+This does not mean sunpy is broken, but you will need to install the extra packages to ensure a "complete" installation of sunpy and run the entire test suite.
+It is quite likely that you will run into not having the tests dependencies installed.

--- a/docs/guide/installation.rst
+++ b/docs/guide/installation.rst
@@ -5,11 +5,7 @@ Installation
 Requirements
 ============
 
-These are the minimum versions of packages needed to install sunpy:
-
-- python 3.7
-- astropy 4.0.3
-- parfive 1.1
+sunpy enforces a minimum Python version that you will have to use and currently that is Python 3.7 or higher.
 
 Installing Scientific Python and sunpy
 ======================================
@@ -21,11 +17,10 @@ If you do not currently have a working scientific Python distribution this guide
 
 To install the Miniconda Python distribution follow the instructions at
 `here <https://docs.conda.io/en/latest/miniconda.html>`__.
-Although Miniconda makes it simple to switch between Python versions, we recommend that new users install
-the latest Python 3.x version of Miniconda as SunPy only supports Python 3.7+.
+Although Miniconda makes it simple to switch between Python versions, we recommend that new users install the latest Python 3.x version of Miniconda.
 
 The reason we choose Miniconda over Anaconda, is mainly due to the size as Anaconda comes with a full install of packages you probably do not need and this way you have more direct control over what has been installed into your Python virtual environment.
-Furthermore, you bypass the need for the conda resolver to sort out your root environment which should make "conda" faster to use.
+Furthermore, you bypass the need for the conda resolver to sort out your root environment which should make conda faster to use.
 
 Installing sunpy using Miniconda
 --------------------------------
@@ -52,8 +47,43 @@ You can update to the latest version by running::
 
     conda update sunpy
 
+Testing sunpy
+=============
+
+sunpy provides a method to run the basic test suite that will check that the install has worked correctly.
+
+To do so, use the :func:`sunpy.self_test`::
+
+    import sunpy
+    sunpy.self_test()
+
+You will see something like the following in your terminal::
+
+    Starting sunpy self test...
+    Checking for packages needed to run sunpy:
+    All required and optional sunpy dependencies are installed.
+    Starting the sunpy test suite:
+    ...
+
+The tests will run and will report any fails which you can report at the `sunpy issue tracker <https://github.com/sunpy/sunpy/issues>`__ and we will strive to help.
+
+It is possible to run this command and in a situation where not all packages are installed in order to run the test suite, you will see::
+
+    Starting sunpy self test...
+    Checking for packages needed to run sunpy:
+    The following packages are not installed for the sunpy[database] requirement:
+    * sqlalchemy
+    ...
+    You do not have all the required dependencies installed to run the sunpy test suite.
+    If you want to run the sunpy tests install the 'tests' extra with `pip install sunpy[all,tests]`
+
+This should not appear under a conda install of sunpy but can under a pip install of sunpy.
+This does not mean sunpy is broken, you will need to install the extra packages to ensure a "complete" installation of sunpy.
+
 Installing sunpy on top of an existing scientific Python environment
 --------------------------------------------------------------------
+
+This section assumes you already have everything setup whether that be Conda or a Python virtual environment and that the commands are exceucted within these.
 
 Conda
 ^^^^^

--- a/setup.cfg
+++ b/setup.cfg
@@ -78,7 +78,6 @@ tests =
   pytest-mock
   pytest-mpl>=0.12 # First version to support our figure tests
   pytest-intercept-remote>=1.2
-  tox
 docs =
   astroquery
   jplephem

--- a/sunpy/tests/self_test.py
+++ b/sunpy/tests/self_test.py
@@ -52,8 +52,9 @@ def self_test(*, package=None, online=False, online_only=False, figure_only=Fals
         print("All required and optional sunpy dependencies are installed.")
     if test_missing:
         print("You do not have all the required dependencies installed to run the sunpy test suite.")
-        print(
-            "If you want to run the sunpy tests install the 'tests' extra with `pip install sunpy[all,tests]`")
+        print(list(test_missing.keys()))
+        print("If are using conda, you will want to run `conda install <package name>`")
+        print("Otherwise you will want run `pip install sunpy[all,tests]`")
         return
     import pytest
     print("Starting the sunpy test suite:")

